src/main/java/org/jvnet/hudson/plugins/repositoryconnector/ArtifactResolver.java

```diff
@@ -124,7 +143,22 @@ private boolean download(AbstractBuild<?, ?> build, BuildListener listener, fina
                 final String targetFileName = TokenMacro.expandAll(build, listener, a.getTargetFileName());
 
                 String version = TokenMacro.expandAll(build, listener, a.getVersion());
+
                 version = checkVersionOverride(build, listener, groupId, artifactId, version);
+                
+                if(version==null) {
+                	version = "LATEST";
+                	/**
+                	List<Version> versions = aether.resolveVersions(groupId, artifactId);
+                	for(Version v : versions) {
+                		logger.println("available version: "+v.getClass().getName()+" / "+v.toString());
+                	}
+                	if(versions.size()>0) {
+                		version = versions.get(versions.size()-1).toString();
+                		logger.println("no version set, resolved version: "+version);
+                	}
+                	**/
```

Comment:
well, that code there would be the alternative to "LATEST", by actually resolving the latest version, which is not working correctly in nexus and discouraged in general (see other comment)